mmv1/third_party/terraform/tests/resource_data_catalog_tag_template_test.go

```diff
@@ -0,0 +1,95 @@
+// ----------------------------------------------------------------------------
+//
+//     ***     AUTO GENERATED CODE    ***    Type: MMv1     ***
+//
+// ----------------------------------------------------------------------------
+//
+//     This file is automatically generated by Magic Modules and manual
+//     changes will be clobbered when the file is regenerated.
+//
+//     Please read more about how to change this file in
+//     .github/CONTRIBUTING.md.
+//
+// ----------------------------------------------------------------------------
+
+package google
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccDataCatalogTagTemplate_dataCatalogTagTemplate_updateFields(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"force_delete":  true,
+		"random_suffix": randString(t, 10),
+	}
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDataCatalogTagTemplateDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataCatalogTagTemplate_dataCatalogTagTemplateBasicExample(context),
+			},
+			{
+				ResourceName:            "google_data_catalog_tag_template.basic_tag_template",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"region", "tag_template_id", "force_delete"},
+			},
+			{
+				Config: testAccDataCatalogTagTemplate_dataCatalogTagTemplateUpdateFields(context),
+			},
+			{
+				ResourceName:            "google_data_catalog_tag_template.basic_tag_template",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"region", "tag_template_id", "force_delete"},
+			},
+		},
+	})
+}
+
+func testAccDataCatalogTagTemplate_dataCatalogTagTemplateUpdateFields(context map[string]interface{}) string {
+	return Nprintf(`
+resource "google_data_catalog_tag_template" "basic_tag_template" {
+  tag_template_id = "tf_test_my_template%{random_suffix}"
+  region = "us-central1"
+  display_name = "Demo Tag Template Test Update"
+
+  fields {
+    field_id = "source"
+    display_name = "Source of data asset test update"
+    type {
```

Comment:
I tested it a bit locally and found out that changing of `type` wi end up with the error like "Error: Error updating TagTemplate Field num_rows: googleapi: Error 400: Cannot change the field type, for field `num_rows`." Do we need a way to handle it? Like adding `ForceNew` to type field?

Reply:
I think, we should rely on the API to return an error for validation. But again, we can handle it by checking the latest type with its last value. Please suggest.

Reply:
I wonder how common the case is for users to change the type a field. However I would lean toward handling the case just as we have the logic of delete + recreate for immutable fields. What's your thoughts?